Mathlib/Data/Set/Image.lean

```diff
@@ -424,9 +424,7 @@ theorem Nonempty.subset_preimage_const {s : Set α} (hs : Set.Nonempty s) (t : S
 
 @[simp]
 theorem preimage_eq_preimage {f : β → α} (hf : Surjective f) : f ⁻¹' s = f ⁻¹' t ↔ s = t :=
-  Iff.intro
-    fun eq => by rw [← image_preimage_eq s hf, ← image_preimage_eq t hf, eq]
-    fun eq => eq ▸ rfl
+  hf.injective_comp_right.eq_iff
```

Comment:
These changes are going to make things painful if we ever want set to become a structure. It might be worth at least grouping the theorems together in a section with an explanatory comment about the defeq abuse in the proof.

Reply:
Though this one should be proved via `hf.preimage_injective.eq_iff` or something.

Reply:
Currently it's the other way: `Surjective.preimage_injective` is proved using this. I can of course change the order.

I'm not aware of situations where making Set a structure would be an improvement. Are you? Nonetheless moving them together is a good idea.